corehq/apps/sms/mixin.py

```diff
@@ -138,26 +142,43 @@ def by_suffix(cls, phone_number, include_pending=False):
         Used to lookup a VerifiedNumber, trying to exclude country code digits.
         """
         try:
-            result = cls.phone_lookup(
-                "phone_numbers/verified_number_by_suffix",
-                phone_number,
-                include_pending
-            )
+            result = cls._by_suffix(apply_leniency(phone_number))
+            return cls._filter_pending(result, include_pending)
         except MultipleResultsFound:
             # We can't pinpoint who the number belongs to because more than one
             # suffix matches. So treat it as if the result was not found.
-            result = None
-        return result
+            return None
+
+    @classmethod
+    @quickcache(['phone_number'], timeout=60 * 60)
+    def _by_phone(cls, phone_number):
+        return cls.phone_lookup('phone_numbers/verified_number_by_number', phone_number)
+
+    @classmethod
+    @quickcache(['phone_number'], timeout=60 * 60)
+    def _by_suffix(cls, phone_number):
+        return cls.phone_lookup('phone_numbers/verified_number_by_suffix', phone_number)
 
     @classmethod
-    def phone_lookup(cls, view_name, phone_number, include_pending=False):
+    def phone_lookup(cls, view_name, phone_number):
```

Comment:
i'm going to be moving the phone numbers over to postgres soon anyway so probably not worth changing it now